pyccel/codegen/utilities.py

```diff
@@ -118,40 +116,42 @@ def compile_files(filename, compiler, flags,
 #                                binary=os.path.splitext(os.path.basename(filename))[0])
 
     o_code = '-o'
-    j_code = ''
+    j_code = []
     if is_module:
-        flags += ' -c '
+        flags.append('-c')
         if (len(output)>0) and language == "fortran":
             if compiler == "ifort":
-                j_code = '-module "{folder}"'.format(folder=output)
+                j_code = ['-module', output]
             else:
-                j_code = '-J"{folder}"'.format(folder=output)
+                j_code = ['-J', output]
 
-    m_code = ' '.join('{}.o'.format(m) for m in modules)
+    m_code = ['{}.o'.format(m) for m in modules]
     if is_module:
-        libs_flags = ''
+        libs_flags = []
     else:
-        flags += ''.join(' -L"{0}"'.format(i) for i in libdirs)
-        libs_flags = ' '.join('-l{}'.format(i) for i in libs)
-
-    filename = '"{}"'.format(filename)  # in case of spaces in path
-    binary = '"{}"'.format(binary)
+        flags.extend(f for i in libdirs for f in ('-L', i))
+        libs_flags = ['-l{}'.format(i) for i in libs]
 
     if sys.platform == "win32" and compiler == "mpif90":
         compiler = "gfortran"
-        filename += ' "{}"'.format(os.path.join(os.environ["MSMPI_LIB64"], 'libmsmpi.a'))
+        m_code.append(os.path.join(os.environ["MSMPI_LIB64"], 'libmsmpi.a'))
 
-    cmd = '{0} {1} {2} {3} {4} {5} {6} {7}'.format( \
-        compiler, flags, m_code, filename, o_code, binary, libs_flags, j_code)
+    cmd = [compiler] + flags + m_code + [filename, o_code, binary] + libs_flags + j_code
```

Comment:
I suspect that Python creates many intermediate `list` objects when processing this line. Shouldn't the following be preferable instead?
```python
cmd = [compiler, *flags, *m_code, filename, o_code, binary, *libs_flags, *j_code]
```

Reply:
Fixed in 4c28592